### PR TITLE
keyboard navigation: feature/focus states

### LINKED ISSUE #575: more visually explicit focus states for focused elements that differs from hover state
Need to finalize design ideas with Lyndsay, going to do a prototype and get approval

Done when the design has been decided on, and implemented.

### PR DESCRIPTION
**Pull request by interim17**
Time estimate or Size
=======
_medium/small_

Problem
=======
Closes #575 and #576, advances #347 
Part of keyboard navigation and accessibility work.

Solution
========
Focus styling for action buttons has been reviewed with Lyndsay but is not reflected in master Figma yet.
Styling for primary/secondary buttons, and dropdown elements is there.
[Style guide](https://www.figma.com/design/u4lZYuVwbsFpzBikgwAVRb/Master-components-Simularium?node-id=1329-1588&node-type=instance&t=0ZsEwbvZECVFUkix-0)

Closing #576 because they already work, its clear how to tab through them now that the visual style is obivous.

Disabled buttons are taking focus/styling for now, this can be discussed more in future as we get into navigation behavior.

* New feature (non-breaking change which adds functionality)

### CustomDropdown
Wrapper for our Dropdowns, unifies the stylesheets, just a slight DRYing now, but will be very useful going forward.

Steps to Verify:
----------------
1. Tab through header and try action buttons (AICS home link, Simularium home link, Download, Share)
2. Open dropdowns and check on hover and focus styling (focus styling is difficult to maintain sometimes with mouse interactions. Easiest way to see it is the use the buttons that open the submenus.

Applying focus styling to ALL interactive elements is out of scope for this PR (e.g. submenu items in drop-downs)

Screenshots (optional):
-----------------------
Screenshot 2024-10-16 at 2 51 56 PM
Screenshot 2024-10-16 at 2 51 50 PM
Screenshot 2024-10-16 at 2 51 43 PM

### CODE REVIEW

File: src/styles/ant-vars.less [hunk 1]
```diff
@@ -73,13 +73,6 @@ https://github.com/ant-design/ant-design/blob/master/components/style/themes/def
 @layout-trigger-background-light: #fff;
 @layout-trigger-color-light     : @text-color;
 
-//Dropdown
-@dropdown-menu-bg: @dark-four;
-@dropdown-vertical-padding: 5px;
-@dropdown-edge-child-vertical-padding: 4px;
-@dropdown-font-size: @font-size-base;
-@dropdown-line-height: 22px;
-
```
Comment (interim17) [inline src/styles/ant-vars.less:82]: Continuing to pivot away from `.less` in anticipation of migrating to antd 5

File: src/components/NavButton/style.css [hunk 2]
```diff
@@ -47,16 +58,28 @@
...
 .nav-button.secondary-button:global(.ant-btn).disabled {
     color: var(--dark-theme-secondary-button-disabled-color);
     background-color: var(--dark-theme-secondary-button-disabled-bg);
     border-color: var(--dark-theme-secondary-button-disabled-border);
     cursor: not-allowed;
+}
+
+.nav-button.dropdown-item-button:global(.ant-btn) {
+    background-color: inherit;
+    border: inherit;
```
Comment (interim17) [inline src/components/NavButton/style.css:84]: In some cases we were using `<a>` to open modals but for accessibility those should be a buttons, even if its easier to style when they are all the same element type. I also felt the styling for all the differenet elements that can be dropdown menu items should be in the `CustomDropdown` stylesheet.

Using `<NavButton />`, making the new `DropdownItem` button class, and using these two inherit rules seemed to be the most concise way to manage selector specificity and keep the styling where I wanted it. Open to suggestions.

File: src/components/CustomDropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            disabled={disabled}
+        >
+            <NavButton
+                titleText={titleText}
+                icon={icon}
+                buttonType={buttonType}
+            />
+        </Dropdown>
+    );
+};
```
Comment (interim17) [inline src/components/CustomDropdown/index.tsx:44]: Like our `CustomModal` and `NavButton` this wrapper is mostly to DRY out the code, and will be a vehicle for code for things that `antd` doesn't provide but can be generalized to all dropdowns.

File: src/components/CustomDropdown/style.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
+.menu {
+    width: var(--dropdown-menu-width);
+    background-color: var(--dark-theme-dropdown-menu-bg);
+    border-radius: 3px;
+    padding: 8px;
+}
+
+.menu.narrow {
+    width: calc(var(--dropdown-menu-width) - 50px);
+}
```
Comment (interim17) [inline src/components/CustomDropdown/style.css:10]: This file replaces the now deleted stylesheets for `HelpMenu` and `LoadFileMenu`.

This was the only style difference between the two to manage for now.

**Review by toloudis — APPROVED** [review]
Taking for granted that everything works, I see no obvious code smells.  LGTM!

File: src/components/HelpMenu/index.tsx [hunk 5]
```diff
@@ -89,26 +87,26 @@ const HelpMenu = (): JSX.Element => {
...
-            <Dropdown menu={{ items, theme: "dark", className: styles.menu }}>
-                <NavButton
-                    titleText={"Help "}
-                    icon={DownArrow}
-                    buttonType={ButtonClass.Secondary}
-                />
-            </Dropdown>
+            <CustomDropdown
+                items={items}
+                titleText={"Help "}
```
Comment (ShrimpCryptid) [inline src/components/HelpMenu/index.tsx:105]: Should `CustomDropdown`/`NavButton` just add spacing between the titleText and the icon?

File: src/components/CustomDropdown/style.css [hunk 6]
```diff
@@ -0,0 +1,48 @@
+.menu {
+    width: var(--dropdown-menu-width);
+    background-color: var(--dark-theme-dropdown-menu-bg);
+    border-radius: 3px;
+    padding: 8px;
+}
+
+.menu.narrow {
+    width: calc(var(--dropdown-menu-width) - 50px);
```
Comment (ShrimpCryptid) [inline src/components/CustomDropdown/style.css:9]: Nit: This seems a little magic number-y, would it make more sense to either define a narrow width variable or make the width set by `fit-content` or something?

File: src/components/LoadFileMenu/index.tsx [hunk 7]
```diff
@@ -117,18 +117,14 @@ const LoadFileMenu = ({
...
                 placement="bottomRight"
                 disabled={isDisabled}
-            >
-                <NavButton
-                    titleText={"Load model"}
-                    icon={DownArrow}
-                    buttonType={ButtonClass.Primary}
-                    isDisabled={isDisabled}
-                />
-            </Dropdown>
```
Comment (ShrimpCryptid) [inline src/components/LoadFileMenu/index.tsx:131]: Really like that the NavButton gets collapsed into Dropdown in all the instances here :')

File: src/components/LoadFileMenu/index.tsx [hunk 8]
```diff
@@ -117,18 +117,14 @@ const LoadFileMenu = ({
 
     return (
         <>
-            <Dropdown
-                menu={{ items, theme: "dark", className: styles.menu }}
```
Comment (ShrimpCryptid) [inline src/components/LoadFileMenu/index.tsx:121]: Do we lose the dark theming here on the dropdown?

File: src/components/NavButton/style.css [hunk 9]
```diff
@@ -6,14 +6,18 @@
...
+    padding: 4px 8px;
 }
 
-.nav-button.action-button:global(.ant-btn:hover),
-.nav-button.action-button:global(.ant-btn:focus) {
+.nav-button.action-button:global(.ant-btn:hover) {
     color: var(--dark-theme-action-button-hover-color);
 }
 
+.nav-button.action-button:global(.ant-btn:focus) {
```
Comment (ShrimpCryptid) [inline src/components/NavButton/style.css:16]: Consider changing the `focus` selectors to `focus-visible`, which will make it only respond to keyboard focusing (and not if the user last clicked on the button)

https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible

**Review by ShrimpCryptid — APPROVED** [review]
Left some non-blocking comments but love the accessibility changes.

File: src/components/LoadFileMenu/index.tsx (hunk 8 above)
Comment (interim17) [reply src/components/LoadFileMenu/index.tsx:121]: No we don't, although we are overriding most of it anyway.

File: src/components/HelpMenu/index.tsx (hunk 5 above)
Comment (interim17) [reply src/components/HelpMenu/index.tsx:105]: Why don't we just remove that silly space instead.

File: src/components/NavButton/style.css (hunk 9 above)
Comment (interim17) [reply src/components/NavButton/style.css:16]: Thank you!

File: src/components/CustomDropdown/style.css (hunk 6 above)
Comment (interim17) [reply src/components/CustomDropdown/style.css:9]: Much better, thank you

File: src/components/NavButton/style.css (hunk 9 above)
Comment (interim17) [reply src/components/NavButton/style.css]: FYI: I updated the lines relevant to this PR , but not every place that `:focus` exists in the code base although I have that in mind now for this ongoing work.

**Pull request merged**